homeassistant/components/recorder/__init__.py

```diff
@@ -273,14 +273,18 @@ def do_purge(now):
                     self.queue.task_done()
                     continue
 
-            with session_scope(session=self.get_session()) as session:
-                dbevent = Events.from_event(event)
-                session.add(dbevent)
-
-                if event.event_type == EVENT_STATE_CHANGED:
-                    dbstate = States.from_event(event)
-                    dbstate.event_id = dbevent.event_id
-                    session.add(dbstate)
+            try:
+                with session_scope(session=self.get_session()) as session:
+                    dbevent = Events.from_event(event)
+                    session.add(dbevent)
+
+                    if event.event_type == EVENT_STATE_CHANGED:
+                        dbstate = States.from_event(event)
+                        dbstate.event_id = dbevent.event_id
+                        session.add(dbstate)
+            except Exception as err:
```

Comment:
- We should catch the specific exception that is being raised.
- We should not mark the event as saved (`self.queue.task_done()`) if we fail? Maybe give it up to X tries?